### PR TITLE
Bug 1877861: fix: return empty Helm repo index file when no HelmChartRepository instances are available in cluster

### PR DESCRIPTION
**Pull request by pedjak**
This fix enables the the requested usecase - an admin should be able to hide Helm chart browsing by removing all instances of `HelmChartRepository` in the cluster.

https://github.com/openshift/console-operator/pull/458 needs to go in first.

### CODE REVIEW

File: pkg/helm/chartproxy/repos.go [hunk 1]
```diff
@@ -188,10 +188,8 @@ func mergeIndexFiles(files ...*repo.IndexFile) *repo.IndexFile {
 func (b *helmRepoGetter) List() ([]*helmRepo, error) {
 	var helmRepos []*helmRepo
 	repos, err := b.Client.Resource(helmChartRepositoryGVK).List(context.TODO(), v1.ListOptions{})
-	if err != nil || len(repos.Items) == 0 {
-		// In case no HelmRepoCRs configured or error reading them, use default redhat helm chart repo
-		helmRepos = append(helmRepos, &DefaultRepo)
```
Comment (jhadvig) [inline pkg/helm/chartproxy/repos.go:193]: if `DefaultRepo` is not used anywhere, it should be removed.

**Review by jhadvig — CHANGES_REQUESTED** [review]

**Review by jhadvig — APPROVED** [review]
/lgtm

File: pkg/helm/chartproxy/repos.go (hunk 1 above)
Comment (pedjak) [reply pkg/helm/chartproxy/repos.go:193]: done.

**Pull request merged**